libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -183,11 +185,39 @@ size_t ContractCompiler::packIntoContractCreator(ContractDefinition const& _cont
 	m_context << deployRoutine;
 
 	solAssert(m_context.runtimeSub() != size_t(-1), "Runtime sub not registered");
+
+	u256 codeMemoryStart = (u256(CompilerUtils::generalPurposeMemoryStart) + m_context.reservedMemory());
```

Comment:
shouldn't the general purpose memory start after the immutable variables?

Reply:
Yep... but currently it's a static constant... we should probably rename it to something like ``specialPurposeMemoryEnd``. The value is only used for initializing the free memory pointer and there I add the memory reserved for immutables on top of it.